chunnel/socket.py

```diff
@@ -51,7 +52,7 @@ def __init__(self, url, params):
 
         transport_class = self.TRANSPORTS[urlsplit(self.url).scheme]
         self.transport = transport_class(
-            self.url, self.params, self._incoming, self._outgoing
+            self.url, self.params, self._incoming, self._outgoing, self.decode_func, self.encode_func
```

Comment:
Think you need to pass the last two as keyword arguments, judging from the test failures.